File: TransCoder_tokenized_test_set_functions/python/COUNT_DISTINCT_ELEMENTS_IN_EVERY_WINDOW_OF_SIZE_K.py
def countDistinct ( arr, k, n ) :
    mp = defaultdict ( lambda : 0 )
    dist_count = 0
    for i in range ( k ) :
        if mp [ arr [ i ] ] == 0 :
            dist_count += 1
        mp [ arr [ i ] ] += 1
    print ( dist_count )
    for i in range ( k, n ) :
        if mp [ arr [ i - k ] ] == 1 :
            dist_count -= 1
        mp [ arr [ i - k ] ] -= 1
        if mp [ arr [ i ] ] == 0 :
            dist_count += 1
        mp [ arr [ i ] ] += 1
        print ( dist_count )
    

File: TransCoder_tokenized_test_set_functions/cpp/COUNT_DISTINCT_ELEMENTS_IN_EVERY_WINDOW_OF_SIZE_K.cpp
void countDistinct ( int arr [ ], int k, int n ) {
  map < int, int > hm;
  int dist_count = 0;
  for ( int i = 0;
  i < k;
  i ++ ) {
    if ( hm [ arr [ i ] ] == 0 ) {
      dist_count ++;
    }
    hm [ arr [ i ] ] += 1;
  }
  cout << dist_count << endl;
  for ( int i = k;
  i < n;
  i ++ ) {
    if ( hm [ arr [ i - k ] ] == 1 ) {
      dist_count --;
    }
    hm [ arr [ i - k ] ] -= 1;
    if ( hm [ arr [ i ] ] == 0 ) {
      dist_count ++;
    }
    hm [ arr [ i ] ] += 1;
    cout << dist_count << endl;
  }
}